Bug: Client crash on quit command

### What part of the program do you think the bug originates from? (frontend, PDF crawler, etc.)

tui client

### Explain the bug

`$ go run . client quit`
Results in
```
panic: Get "http://localhost:8080/quit": EOF

goroutine 1 [running]:
seekourney/client.checkHTTPError(...)
	/home/oxygen/Projects/Seekourney-Weaver/client/client.go:64
seekourney/client.shutdownServer()
	/home/oxygen/Projects/Seekourney-Weaver/client/client.go:102 +0x45
seekourney/client.Run({0xc000014160?, 0x7470e0?, 0xc000060230?})
	/home/oxygen/Projects/Seekourney-Weaver/client/client.go:55 +0xb3
main.main()
	/home/oxygen/Projects/Seekourney-Weaver/main.go:20 +0x65
exit status 2
```

### If it is not clear how to reproduce the bug please add more information here

I do not have for time for this right now

### Add any misc info here

_No response_

--- a/backend/core/server/server.go
+++ b/backend/core/server/server.go
@@ -91,24 +91,25 @@ func Run(args []string) {
 		BaseContext: func(l net.Listener) context.Context { return ctx },
 	}
 
-	http.HandleFunc("/", func(writer http.ResponseWriter, request *http.Request) {
-		serverParams := serverFuncParams{
-			writer: writer,
-			db:     db,
-			stop:   stop,
-		}
+	http.HandleFunc("/",
+		func(writer http.ResponseWriter, request *http.Request) {
+			serverParams := serverFuncParams{
+				writer: writer,
+				db:     db,
+				stop:   stop,
+			}
 
-		switch html.EscapeString(request.URL.Path) {
-		case "/all":
-			handleAll(serverParams)
-		case "/search":
-			handleSearch(serverParams, request.URL.Query()["q"])
-		case "/add":
-			handleAdd(serverParams, request.URL.Query()["p"])
-		case "/quit":
-			handleQuit(serverParams)
-		}
-	})
+			switch html.EscapeString(request.URL.Path) {
+			case "/all":
+				handleAll(serverParams)
+			case "/search":
+				handleSearch(serverParams, request.URL.Query()["q"])
+			case "/add":
+				handleAdd(serverParams, request.URL.Query()["p"])
+			case "/quit":
+				handleQuit(serverParams)
+			}
+		})
 
 	go func() {
 		err := server.ListenAndServe()
@@ -127,7 +128,11 @@ func Run(args []string) {
 	if err != nil {
 		fmt.Println("Error while shutting down server: ", err)
 	}
-	db.Close()
+	err = db.Close()
+	if err != nil {
+		fmt.Println("Error while closing database: ", err)
+	}
+
 	stopContainer()
 }
 

